src/coffee/io/reader.coffee

```diff
@@ -0,0 +1,107 @@
+_ = require 'underscore'
+Csv = require 'csv'
+path = require 'path'
+Promise = require 'bluebird'
+iconv = require 'iconv-lite'
+fs = Promise.promisifyAll require('fs')
+Excel = require 'exceljs'
+
+debugLog = console.log
+debugLog = _.noop
+
+
+class Reader
+
+  constructor: (@options = {}) ->
+    debugLog "READER::options:", JSON.stringify(@options)
+    @options.encoding = @options.encoding || 'utf-8'
+    @options.csvDelimiter = @options.csvDelimiter || ','
+    @header = null
+    @rows = []
+
+    if not iconv.encodingExists @options.encoding
+      throw new Error 'Encoding does not exist: ' + @options.encoding
+
+  read: (file) =>
+    debugLog "READER::stream file %s", file
+    @inputStream = fs.createReadStream file
```

Comment:
Stream can be used by all fileParsing services (xlsx, csv) and we can later stream (for example) also data from STDIN or transform/clean data somehow. It was originally loading a whole dataset before processing so I kept it that way.